File: Quizify-FrontEnd/src/app/quizify/tsclasses/game.ts
import { Question } from './question';
import { Category } from './category';
import { Topic } from './topic';
import { Genre } from './genre';
import { Tag } from './tag';

export class Game {
    id: number;
    name: string;
    category: Category;
    topic: Topic;
    genre: Genre;
    level: string;
    imageUrl: string;
    numOfQuestion: number;
    questions: Question[];
    timeDuration: number;
    liked: number;
    playCount: number;
    rules: string[];
    pointPerQuestion: number;
    totalPoints: number;
    playerScore: number;
}

File: single-player-engine/src/main/java/com/stackroute/quizify/singleplayerengine/domain/Game.java
package com.stackroute.quizify.singleplayerengine.domain;


import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.List;



@Data
public class Game {
    @Id
    private long id;
    private String name;
    private Category category;
    private Topic topic;
    private Genre genre;
    private String level;
    private String imageUrl;
    private int numOfQuestion;
    private List<Question> questions;
    private int timeDuration;
    private int liked;
    private int playCount;
    private List<String> rules;
    private int pointPerQuestion;
    private int totalPoints;
    private int playerScore;



}
